### PR TITLE
Make Nginx proxy with HTTP/1.1 for Licensing

### PR DESCRIPTION
**Pull request by alexmuller**
This commit adds a feature flag so that Nginx on the app servers can proxy to the Licensing app on localhost using HTTP/1.1.

Nginx docs:

> By default, version 1.0 is used. Version 1.1 is recommended for use with keepalive connections and NTLM authentication.

http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_http_version

Licensing has been upgraded to a new version of Scala/Play which is picky about which HTTP version it needs (because it uses chunked responses for file downloads).

### CODE REVIEW

File: modules/govuk/manifests/app.pp [hunk 1]
```diff
@@ -253,6 +253,10 @@
 # Configure the amount of time the nginx proxy vhost will wait for the
 # backing app before it sends the client a 504. It defaults to 15 seconds.
 #
+# [*proxy_http_version_1_1_enabled*]
+#   Boolean, whether to enable HTTP/1.1 for proxying from the Nginx vhost
+#   to the app server.
```
Comment (elliotcm) [inline modules/govuk/manifests/app.pp]: Could you add the default to this (in another PR if you like)

File: modules/govuk/manifests/app/config.pp [hunk 2]
```diff
@@ -12,6 +12,10 @@
 # [*nagios_memory_critical*]
 #   Memory use, in MB, at which Nagios should generate a critical alert.
 #
+# [*proxy_http_version_1_1_enabled*]
+#   Boolean, whether to enable HTTP/1.1 for proxying from the Nginx vhost
+#   to the app server.
```
Comment (elliotcm) [inline modules/govuk/manifests/app/config.pp]: Ditto default

File: modules/nginx/manifests/config/vhost/proxy.pp [hunk 3]
```diff
@@ -97,6 +101,7 @@
   $ensure = 'present',
   $alert_5xx_warning_rate = 0.05,
   $alert_5xx_critical_rate = 0.1,
+  $proxy_http_version_1_1_enabled = false,
```
Comment (elliotcm) [inline modules/nginx/manifests/config/vhost/proxy.pp]: Is this passed automatically to the template?

**Comment by elliotcm** [comment]
Do we need to enable this on the feed app?

File: modules/govuk/manifests/app.pp (hunk 1 above)
Comment (alexmuller) [inline modules/govuk/manifests/app.pp]: Sure, will do it in another pull request

File: modules/nginx/manifests/config/vhost/proxy.pp (hunk 3 above)
Comment (alexmuller) [inline modules/nginx/manifests/config/vhost/proxy.pp]: Yep, it'll go through

**Comment by alexmuller** [comment]
I'm not sure about the feed app but as it has a vhost and it's the same Scala app I don't think it will hurt.

**Comment by elliotcm** [comment]
👍 when green
